debezium-core/src/main/java/io/debezium/relational/history/TableChanges.java

```diff
@@ -36,11 +36,16 @@ public TableChanges() {
         this.changes = new ArrayList<>();
     }
 
-    public static TableChanges fromArray(Array array) {
+    /*
+     * useCatalogBeforeSchema is true if the parsed string contains only 2 items and the first should be used as
+     * the catalog and the second as the table name, or false if the first should be used as the schema and the
+     * second as the table name
+     */
+    public static TableChanges fromArray(Array array, boolean useCatalogBeforeSchema) {
```

Comment:
I would have expected that the `toArray()` method needs adjustment, too, for the sake of symmetry between serializing and parsing back?

Reply:
Not really, as far as I can tell at least: when the TableId gets serialized in the toArray() method, an "id" string is built and information is lost on how the id was obtained (depending on whether there was a catalog, schema and table defined). To deserialize the TableId, the information is recovered with the useCatalogBeforeSchema parameter. Now, one could standardize how TableIds are serialized, so that parsing is possible without external information, but wanted to minimize changes to the core. I can't exclude that I may be missing something though :)

Reply:
Ah yes, makes sense, thanks!